storage/response_cache.go

```diff
@@ -90,14 +91,16 @@ func (rc *ResponseCache) GetResponse(cmdID proto.ClientCmdID, reply proto.Respon
 	// If the response is in the cache or we experienced an error, return.
 	rwResp := proto.ReadWriteCmdResponse{}
 	key := keys.ResponseCacheKey(rc.raftID, &cmdID)
-	if ok, err := engine.MVCCGetProto(rc.engine, key, proto.ZeroTimestamp, true, nil, &rwResp); ok || err != nil {
-		if err == nil && rwResp.GetValue() != nil {
-			gogoproto.Merge(reply.(gogoproto.Message), rwResp.GetValue().(gogoproto.Message))
+	ok, err := engine.MVCCGetProto(rc.engine, key, proto.ZeroTimestamp, true, nil, &rwResp)
+	if ok && err == nil {
+		value := rwResp.GetValue()
+		if respProto, isMsg := value.(gogoproto.Message); isMsg {
+			gogoproto.Merge(reply, respProto)
+		} else {
+			panic(fmt.Sprintf("value %s does not implement gogoproto.Message", value))
```

Comment:
6 of one, half-dozen of the other - I'd rather panic than return an error because we never expect this to happen, and I like the explicitness of this compared to the unchecked cast. reverted to an unchecked cast